src/BmmPlans.tsx

```diff
@@ -0,0 +1,227 @@
+                            <TextField required id="b1" style={{width: 60}} defaultValue="-30" />&nbsp;
+                            <TextField required id="b2" style={{width: 60}} defaultValue="-10" />&nbsp;
+                            <TextField required id="b3" style={{width: 60}} defaultValue="15.5" />&nbsp;
+                            <TextField required id="b4" style={{width: 60}} defaultValue="15k" />
+                        </div>
+                        <div>
+                            <FormLabel component="legend">Steps:</FormLabel>
+                            <TextField required id="b0" style={{width: 60}} defaultValue="10" />&nbsp;
+                            <TextField required id="b1" style={{width: 60}} defaultValue="2.0" />&nbsp;
+                            <TextField required id="b2" style={{width: 60}} defaultValue="0.3" />&nbsp;
+                            <TextField required id="b4" style={{width: 60}} defaultValue="0.05k" />
+                        </div>
+                        <div>
+                            <FormLabel component="legend">Times:</FormLabel>
+                            <TextField required id="b0" style={{width: 60}} defaultValue="0.5" />&nbsp;
+                            <TextField required id="b1" style={{width: 60}} defaultValue="0.5" />&nbsp;
+                            <TextField required id="b2" style={{width: 60}} defaultValue="0.5" />&nbsp;
+                            <TextField required id="b4" style={{width: 60}} defaultValue="0.25k" />
+                        </div><br />
+                        <Button variant="contained">Submit</Button>
+                    </form>
+                </Grid>   
+                <Grid item justify="center" spacing={10} xs={3}>
+                  <PlanList clearQueue={this.props.clearQueue} plans={this.props.plans}></PlanList> 
+                </Grid>
+            </Grid>
+          </Container>
+        )
+    }
+
+    private handleSelectPlan = (selectedPlan: string) => {
+        this.setState({ selectedPlan });
+    };
+
+    private handleParamChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+        this.state.onPlanParamChange(event.target.value as number);
+    };
+
+    private handlePlanParamChange = (planParam: number) => {
+        this.setState({ planParam });
+    };
+
+    private handleEnvChange = (env: string) => {
+        this.setState({ env });
+    };
+
+    private handleQueueChange = (queue: string) => {
+        this.setState({ queue });
+    };
+
+    private handleSubmitClick = () => {
+        this.props.submitPlan(this.state.selectedPlan, this.state.planParam);
+    }
+
+    private handleEnvClick = () => {
+        if (this.state.env === "Open") {
+            this.props.modifyEnvironment(EnvOps.open);
+            this.state.onEnvChange("Close");
+        }
+        else {
+            this.props.modifyEnvironment(EnvOps.close);
+            this.state.onEnvChange("Open");
+        }
+    }
```

Comment:
It seems to be a copy-pasted method. Can it be reused from somewhere where it already exists?